### PR TITLE
Upgrade to selenium 4

### PR DESCRIPTION
**Pull request by mkousteris**
## 📝 Description

* CHG: Upgrade to latest selenium client v4.4.0
* ADD: Generic method for getting WebDriver with typeCast
* ADD: ClientConfig to RemoteWebDriver instantiation along with readTimeout and retries

* FIX: issue with not taken screenshot
* FIX: issues with stuck WebDriver socket connections by overwriting readTimeout after instantiating RemoteWebDriver - Example of screenshot taken when WebDriver's windowHandle is different to active browser tab
* FIX: issue with findChildElement based on Sizzle css selectors

* RMV: not used Applitools dependency and implementation
* RMV: CompletableFutureTask resulting in multiple spawned nodes in selenium Grid

## 🛂 Type of change

- [X] New feature (non-breaking change which adds functionality)

## ℹ️ Additional comments (if any)

Final regression pending

### CODE REVIEW

**Review by ptsiakos77 — APPROVED** [review]
excellent 🎉

File: src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java [hunk 1]
```diff
@@ -102,23 +111,34 @@ public String getBeanName() {
         return "webDriverController";
     }
 
-    private WebDriver getRemoteWebDriver(String rcUrl, Capabilities desiredCapabilities) {
-        WebDriver driver;
-        Augmenter augmenter = new Augmenter(); // adds screenshot capability to a default webdriver.
+    private WebDriver getRemoteWebDriver(String rcUrl, Capabilities desiredCapabilities) throws MalformedURLException, NoSuchFieldException, IllegalAccessException {
+        WebDriver driver = null;
+        //ReadTimeout when creating the RemoteWebDriver objects
+        //needs to take into consideration the time needed for
```
Comment (taniapil) [inline src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java]: Are these comments needed. I think you're explaining the same thing in the block below

File: src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java [hunk 2]
```diff
@@ -102,23 +111,34 @@ public String getBeanName() {
...
+        /**
+         * When setting readTimeout RemoteWebDriver creation
+         * we need to take into consideration the time needed for the Grid node to be spawned
+         * Gridlastic suggests setting it to 10 minutes
+         */
+        ClientConfig config = ClientConfig.defaultConfig().baseUrl(new URL(rcUrl)).readTimeout(Duration.ofMinutes(Integer.parseInt(SteviaContext.getParam("nodeTimeout")))).withRetries();
+        Tracer tracer = OpenTelemetryTracer.getInstance();
+        CommandExecutor executor = new HttpCommandExecutor(Collections.emptyMap(), config, new TracedHttpClient.Factory(tracer, org.openqa.selenium.remote.http.HttpClient.Factory.createDefault()));
```
Comment (taniapil) [inline src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java:123]: I can't say I understand what you're doing here but I trust you 😛

File: src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/webdriverapi/BySizzle.java [hunk 3]
```diff
@@ -38,115 +38,68 @@
...
-     * @param xpathExpression The xpath to use
-     * @return a By which locates elements via XPath
-     */
-    public static By xpath(final String xpathExpression) {
-        if (xpathExpression == null)
-            throw new IllegalArgumentException(
-                    "Cannot find elements when the XPath expression is null.");
-
-        return new ByXPathExtended(xpathExpression);
-    }
```
Comment (taniapil) [inline src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/webdriverapi/BySizzle.java:85]: Why did you remove the xpath related code? Are the xpath selectors handled in another class?

**Review by taniapil — APPROVED** [review]

**Comment by taniapil** [comment]
Great work 💯

File: src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/webdriverapi/BySizzle.java (hunk 3 above)
Comment (mkousteris) [reply src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/webdriverapi/BySizzle.java:85]: This class is only used for sizzle related locators, for xpath native By class is used
https://github.com/Workable/stevia/blob/634b3edd2e823c09e96cf49d4b8358003c7b0a6b/src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/WebDriverWebController.java#L239

File: src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java (hunk 1 above)
Comment (mkousteris) [reply src/main/java/com/persado/oss/quality/stevia/selenium/core/controllers/factories/WebDriverWebControllerFactoryImpl.java]: Removed
